common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/SimplePostgresqlClient.java

```diff
@@ -0,0 +1,173 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql;
+
+import org.eclipse.dataspaceconnector.clients.postgresql.connection.ConnectionFactory;
+import org.eclipse.dataspaceconnector.clients.postgresql.row.RowMapper;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.ArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.BigDecimalArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.BooleanArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.ByteArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.BytesArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.DateArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.DoubleArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.FloatArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.InputStreamArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.IntArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.LongArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.NullArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.ObjectArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.ShortArgumentHandler;
+import org.eclipse.dataspaceconnector.clients.postgresql.statement.StringArgumentHandler;
+import org.jetbrains.annotations.NotNull;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class SimplePostgresqlClient implements PostgresqlClient {
```

Comment:
I partially agree. Since the `pooled` implementation relies on the `simple` one (due to scoping a transaction to a specific connection), we cannot remove it completely. But the `SimplePostgresqlClient` could be made package private and the `PooledPostgresqlClient` be ranmed to e.g. `PostgresqlClientImpl`.

Reply:
`SimplePostgresqlClient` has been removed and `PooledPostgresqlClient` was renamed to `PostgresqlClientImpl`.